apps/wwaw/build_wwaw.py

```diff
@@ -0,0 +1,604 @@
+from urllib.parse import urljoin, urlparse
+from time import sleep
+from argparse import ArgumentParser
+from tldextract import extract
+from random import uniform
+from random import choices
+from string import ascii_lowercase
+from string import digits
+from requests import get
+from requests import RequestException
+from os import makedirs
+from hashlib import md5
+from re import sub
+from bs4 import BeautifulSoup
+from typing import List
+from typing import Tuple
+from typing import Optional
+from hocon_constants import HOCON_HEADER_REMAINDER
+from hocon_constants import HOCON_HEADER_START
+from hocon_constants import LEAF_NODE_AGENT_TEMPLATE
+from hocon_constants import REGULAR_AGENT_TEMPLATE
+from hocon_constants import TOP_AGENT_TEMPLATE
+
+# Regex to replace all non-alphanumeric and non-hyphen characters with an empty string
+SAFE_AGENT_NAME_CHARS_REGEX = r"[^a-zA-Z0-9\-]"
+
+# Regex to replace all non-word, non-space, and non-hyphen characters with a space
+AGENT_NAME_SANITIZE_REGEX = r"[^\w\s-]"
+
+# Regex to replace sequences of whitespace or underscores with a single hyphen
+AGENT_NAME_HYPHENATE_REGEX = r"[\s_]+"
+
+# Regex to remove URLs from extracted text
+URL_REGEX = r"https?://\S+"
+
+# Regex to remove scene7 junk or custom format @(...) from extracted text
+SCENE7_JUNK_REGEX = r"@\(.*?\)"
+
+
+class WebAgentNetworkBuilder:
+    TOTAL_AGENTS = 40
+    MAX_CHILDREN = 10
+    MAX_NAME_LEN = 40  # Cannot be more than 55
+    PAGE_LEN_MAX = 5000
+    MIN_PAGE_LEN = 200
+    START_URL = "https://www.cognizant.com/us/en"
+    AGENT_NETWORK_NAME = f"autogenerated_agent_network_{TOTAL_AGENTS}"
+    OUTPUT_PATH = "../../registries/"  # Make sure the new hocon is added to the manifest
+    AGENT_INSTRUCTION_PREFACE = ("You represent the following content from a web page for the user and can answer "
+                                 "questions related to this content. "
+                                 "The web page you represent is part of a larger web site, represented by your tools. "
+                                 "Always use your tools as an extension of your content. Your content might not be up "
+                                 "to date or complete, but your tools may have the latest or complementary information,"
+                                 " so always check with your tools too. "
+                                 "Make sure you give the user all the information you have in all the content you and "
+                                 "your tools represent."
+                                 "Here is the specific content that you represent and the remainder is represented by "
+                                 "your tools: ")
+
+    def __init__(self):
+        self.agent_counter = 0
+        self.politeness_delay = 0.0
+        self.top_agent_name = None
```

Comment:
Yay! State that you don't have to pass around all over the place in method args!